### DISCUSSION TITLE
refresh GAP8 of AI deck by OPENOCD

### DISCUSSION THREAD

**Discussion by coo122567** [bitcraze/discussions#416] [Q&A]
Hello everyone,
how to install OPENOCD on Windows?
Because I saw that refreshing GAP8 needs to use JTAG and also use OPNOCD to compile. 
I am a novice, so I would like to ask everyone, thank you

**Comment by evoggy** [comment]
Hi,

I would recommend using Linux (preferably Ubuntu) for this. Please see [these instructions](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#gap8-bootloader-for-the-aideck-1-0-or-aideck-1-1-rev-a-c-) on how to flash it.

**Reply by coo122567** [reply]
Hello, can I download the desktop version of docker on windows to use? 
Because the current computer is windows instead of linux 。

**Reply by knmcguire** [reply]
yes docker also exist for Windows, but we don't recommend development for the AIdeck on windows. Consider using [ the virtual machine ](https://github.com/bitcraze/bitcraze-vm) for that or make one for Ubuntu yourselve. or WSL2 on windows as long as you use the over air flashing (programmer flashing connection I've had trouble myself with).
